hs-bindgen/app/HsBindgen/Cli/Info/Libclang.hs

```diff
@@ -58,9 +61,40 @@ exec :: GlobalOpts -> Opts -> IO ()
 exec GlobalOpts{..} Opts{..} =
     void . withTracer tracerConfig $ \tracer _ -> do
       clangArgs <- getClangArgs (contramap TraceBoot tracer) clangArgsConfig
-      let setup = defaultClangSetup clangArgs $
+      let hasNoUserOptions = hasNoUserClangOptions clangArgsConfig
+          setup = defaultClangSetup clangArgs $
             ClangInputMemory "hs-bindgen-nop.h" ""
+
+      -- Emit informational message if no user options provided
+      when (   hasNoUserOptions
+            && unwrapVerbosity (tVerbosity tracerConfig) >= Info) $ do
+        traceWith (contramap (TraceFrontend . FrontendClang) tracer)
+                  ClangInvokedWithoutOptions
+
+      -- Use a filtering tracer that suppresses CXError_ASTReadError.
+      -- This error is expected when diagnostic options like -v or
+      -- -print-target-triple are used, as Clang exits early without
+      -- creating a full AST after printing the requested information.
+      --
+      let clangTracer = squelchUnless (not . isASTReadError)
```

Comment:
Is the `isASTReadError` a trace (or is it simply output from `libclang`)? If it is a trace, this can be done in a simpler way by using `CustomLogLevel`.

There are some examples (`CustomLogLevelSetting`) in `TraceMsg.hs`.